optuna/storages/grpc/api.proto

```diff
@@ -0,0 +1,440 @@
+message GetStudySystemAttributesRequest {
+    int64 study_id = 1;
+}
+
+/**
+ * Reply to get study system attributes.
+ */
+message GetStudySystemAttributesReply {
+    map<string, string> system_attributes = 1;
+}
+
+/**
+ * Request to get all studies.
+ */
+message GetAllStudiesRequest {}
+
+/**
+ * Reply to get all studies.
+ */
+message GetAllStudiesReply {
+    repeated Study studies = 1;
+}
+
+/**
+ * Request to create a new trial.
+ */
+message CreateNewTrialRequest {
+    int64 study_id = 1;
+    Trial template_trial = 2;
+    bool template_trial_is_none = 3;
+}
+
+/**
+ * Reply to create a new trial.
+ */
+message CreateNewTrialReply {
+    int64 trial_id = 1;
+}
+
+/**
+ * Request to set a trial parameter.
+ */
+message SetTrialParameterRequest {
+    int64 trial_id = 1;
+    string param_name = 2;
+    double param_value_internal = 3;
+    string distribution = 4;
+}
+
+/**
+ * Reply to set a trial parameter.
+ */
+message SetTrialParameterReply {}
+
+/**
+ * Request to get a trial id from its study id and trial number.
+ */
+message GetTrialIdFromStudyIdTrialNumberRequest {
+    int64 study_id = 1;
+    int64 trial_number = 2;
+}
+
+/**
+ * Reply to get a trial id from its study id and trial number.
+ */
+message GetTrialIdFromStudyIdTrialNumberReply {
+    int64 trial_id = 1;
+}
+
+/**
+ * Request to set trial state and values.
+ */
+message SetTrialStateValuesRequest {
+    int64 trial_id = 1;
+    TrialState state = 2;
+    repeated double values = 3;
+}
+
+/**
+ * Reply to set trial state and values.
+ */
+message SetTrialStateValuesReply {
+    bool trial_updated = 1;
+}
+
+/**
+ * Request to set a trial intermediate value.
+ */
+message SetTrialIntermediateValueRequest {
+    int64 trial_id = 1;
+    int64 step = 2;
+    double intermediate_value = 3;
+}
+
+/**
+ * Reply to set a trial intermediate value.
+ */
+message SetTrialIntermediateValueReply {}
+
+/**
+ * Request to set a trial user attribute.
+ */
+message SetTrialUserAttributeRequest {
+    int64 trial_id = 1;
+    string key = 2;
+    string value = 3;
+}
+
+/**
+ * Reply to set a trial user attribute.
+ */
+message SetTrialUserAttributeReply {}
+
+/**
+ * Request to set a trial system attribute.
+ */
+message SetTrialSystemAttributeRequest {
+    int64 trial_id = 1;
+    string key = 2;
+    string value = 3;
+}
+
+/**
+ * Reply to set a trial system attribute.
+ */
+message SetTrialSystemAttributeReply {}
+
+/**
+ * Request to get a trial by its ID.
+ */
+message GetTrialRequest {
+    int64 trial_id = 1;
+}
+
+/**
+ * Reply to get a trial by its ID.
+ */
+message GetTrialReply {
+    Trial trial = 1;
+}
+
+/**
+ * Request to get all trials in a study.
+ */
+message GetAllTrialsRequest {
+    int64 study_id = 1;
+    repeated TrialState states = 2;
+}
+
+/**
+ * Reply to get all trials in a study.
+ */
+message GetAllTrialsReply {
+    repeated Trial trials = 1;
+}
+
+/**
+ * Study direction.
+ */
+enum StudyDirection {
+    MINIMIZE = 0;
+    MAXIMIZE = 1;
+}
```

Comment:
I think this comment has not been applied.
https://github.com/optuna/optuna/pull/5852#discussion_r1877542635

<img width="821" alt="image" src="https://github.com/user-attachments/assets/8bf63f76-5e6e-45fb-a18e-4aa2d82ad4ed" />

Reply:
Thanks for the reminder. I searched the use cases of `StudyDirection.NOT_SET`, but I don't see any ones as follows.
```
> git grep NOT_SET
optuna/storages/_rdb/alembic/versions/v0.9.0.a.py:            sa.Enum("NOT_SET", "MINIMIZE", "MAXIMIZE", name="studydirection"),
optuna/storages/_rdb/alembic/versions/v2.4.0.a.py:                sa.Enum("NOT_SET", "MINIMIZE", "MAXIMIZE", name="studydirection"),
optuna/study/_study_direction.py:        NOT_SET:
optuna/study/_study_direction.py:    NOT_SET = 0
```
In addition, the `StudyDirection.NOT_SET` is not allowed in the `create_study`([ref](https://github.com/optuna/optuna/blob/master/optuna/study/study.py#L1251)). So, I think we don't have to introduce it in the gRPC proxy.